modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/classic_richardson.h

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "modules/linear_boltzmann_solvers/lbs_solver/iterative_methods/wgs_context.h"
+#include "modules/linear_boltzmann_solvers/lbs_solver/groupset/lbs_groupset.h"
+#include "framework/math/linear_solver/linear_solver.h"
+#include <memory>
+#include <vector>
+#include <functional>
+
+namespace opensn
+{
+namespace lbs
+{
+
+/**Linear Solver specialization for Within GroupSet (WGS) solves with classic
+ * Richardson.
+ */
+class ClassicRichardson : public LinearSolver
+{
+public:
+  /**
+   * Constructor.
+   * \param gs_context_ptr Context Pointer to abstract context.
+   */
+  explicit ClassicRichardson(std::shared_ptr<WGSContext> gs_context_ptr);
+
+  ~ClassicRichardson() override;
+
+  void Setup() override {}
+
+  void Solve() override;
+
+protected:
+  void PreSetupCallback() override {}
+
+  void SetConvergenceTest() override {}
+
+  void SetSystemSize() override {}
+
+  void SetSystem() override {}
+
+  void SetPreconditioner() override {}
+
+  void PostSetupCallback() override {}
+
+  void PreSolveCallback() override {}
+
+  void SetRHS() override {}
+
+  void SetInitialGuess() override {}
+
+  void PostSolveCallback() override {}
+
+private:
+  std::vector<double> saved_q_moments_local_;
+  std::vector<double> psi_new_, psi_old_;
```

Comment:
If I am reading this correctly, now there is `psi_new_local` in `LBSSolver` and then `psi_new` and `psi_old` per groupset? Do the PETSC solvers converge on angular fluxes? Should they? If not, why are we doing this here?

Reply:
I can probably get rid of `psi_new_` here and use `psi_new_local` in `LBSSolver` instead. I'll see about making that change. The PETSc solvers do converge on angular fluxes... which is the appropriate thing to do.

Reply:
I've opted against using `psi_new_local` in `LBSSolver` for this purpose for a couple of reasons. First, it would prevent the use of classic Richardson with time-dependent or adjoint solvers. Maybe not prevent, but it would be very inefficient. Second, I'm not sure we want to have psi at the level of `LBSSolver` or even at the discrete ordinates solver level. It seems to me this should be a private member for each individual solver (adjoint, time-dependent, etc.). I think we need to look more closely at how we want to handle storing psi as part of the refactoring of `LBSSolver`.